python/pyspark/sql/tests.py

```diff
@@ -2993,6 +2990,7 @@ def test_current_database(self):
             AnalysisException,
             "does_not_exist",
             lambda: spark.catalog.setCurrentDatabase("does_not_exist"))
+        spark.sql("DROP DATABASE some_db")
```

Comment:
create and drop should be part of test case,if there is any exception then test case will fail.So no need to put in the finally block

Reply:
We still need to cleanup each test. Otherwise, if the test fails before dropping `some_db`, it will cause to fail other tests.
Seems like the current cleanup is not good. Let me refine.

Reply:
Submitted #22762.